tests/unit/sanakoyeu_et_al_2018/test_loss.py

```diff
@@ -1,7 +1,184 @@
+                ptu.assert_allclose(op.accuracy, desired_accuracy)
+
+
+def test_MultiLayerPredictionOperator(subtests, input_image):
+    def get_encoding_op(encoder, score_weight):
+        return TestOperator(encoder, score_weight)
+
+    layers = [str(index) for index in range(3)]
+    modules = [(layer, nn.Conv2d(3, 3, 1)) for layer in layers]
+    multi_layer_encoder = enc.MultiLayerEncoder(modules)
+
+    multi_layer_prediction_op = paper.MultiLayerPredictionOperator(
+        multi_layer_encoder, layers, get_encoding_op
+    )
+
+    for mode in (True, False):
+        multi_layer_prediction_op.real() if mode else multi_layer_prediction_op.fake()
+        _ = multi_layer_prediction_op(input_image)
+        with subtests.test(mode=mode):
+            desired = torch.mean(
+                torch.stack(
+                    [
+                        op.accuracy
+                        for op in multi_layer_prediction_op.discriminator_operators()
+                    ]
+                )
+            )
+            ptu.assert_allclose(multi_layer_prediction_op.get_accuracy(), desired)
+
+
+def test_prediction_loss(subtests):
+    for impl_params in (True, False):
+        prediction_loss = paper.prediction_loss(impl_params=impl_params,)
+        assert isinstance(prediction_loss, paper.MultiLayerPredictionOperator)
+
+        with subtests.test("encoding_ops"):
+            assert all(
+                isinstance(op, paper.PredictionOperator)
+                for op in prediction_loss.discriminator_operators()
+            )
+
+        predictors, layers, layer_weights = zip(
+            *[
+                (op.predictor, op.encoder.layer, op.score_weight)
+                for op in prediction_loss.discriminator_operators()
+            ]
+        )
+
+        with subtests.test("predictor"):
+            for predictor, kernel_size in zip(predictors, (5, 10, 10, 6, 3)):
+                assert isinstance(predictor, nn.Module)
+                assert predictor.kernel_size == misc.to_2d_arg(kernel_size)
+
+        with subtests.test("layers"):
+            desired_layers = {"0", "1", "3", "5", "6"}
+            assert set(layers) == desired_layers
+
+        with subtests.test("layer_weights"):
+            desired_layer_weights = (1e0,) * len(desired_layers)
+            assert layer_weights == pytest.approx(desired_layer_weights)
+
+        with subtests.test("score_weight"):
+            score_weight = 1e0 if impl_params else 1e-3
+            assert prediction_loss.score_weight == pytest.approx(score_weight)
+
+
+def attach_method_mock(mock, method, **attrs):
+    if "name" not in attrs:
+        attrs["name"] = f"{mock.name}.{method}()"
+
+    method_mock = unittest.mock.Mock(**attrs)
+    mock.attach_mock(method_mock, method)
+
+
+@pytest.fixture
+def prediction_loss_mocks(mocker):
+    attrs = {}
+    attrs["side_effect"] = lambda image: pystiche.LossDict([("0", torch.mean(image))])
+
+    mock = mocker.Mock(**attrs)
```

Comment:
Why this weird initalization?

```suggestion
    mock = mocker.Mock(side_effect=lambda image: pystiche.LossDict([("0", torch.mean(image))]))
```